### DISCUSSION TITLE
Roll-your-own command line argument parser rather than library

### DISCUSSION THREAD

**Discussion by david-bakin** [vector-of-bool/dds#137] [General]
I notice you rolled-your-own command line argument parser rather than using a library (it looks nice too by the way) - even though `repo-3.bpt.pizza` is filled with nice available OSS libraries for everything from yaml/toml/json decode/encode to logging to some reflection stuff to date handling to compile-time-RE etc etc etc.

So I'm curious (and this is the "general" topic so why not ask here): Why didn't you pick one of the many "modern c++" argument parsing libraries out there?  (And make it available via the bpt repo?) Specifically - what kind of _drawbacks_ are there to the existing ones that made you  prefer one of your own creation?

**Comment by vector-of-bool** [comment]
I have a strong background in Python as well, and I've very often used the argparse module for writing command-line programs. I've found almost all existing solutions to be lackluster in comparison, especially when it comes to subparsers. Other than that, there is no particular reason that I rolled my own. The intent is that I will pull certain aspects of bpt out into their own library, with the command parser being one of them.
